src/context/auth/auth.test.tsx

```diff
@@ -131,33 +125,29 @@ describe("auth", () => {
       const mockFetchPromise = jest.fn().mockResolvedValue({});
       jest.spyOn(global, "fetch").mockImplementation(mockFetchPromise);
 
-      const { Component, hook } = renderComponentWithHook();
-      const { history } = render(<Component />, { wrapper });
-
-      act(() => {
-        hook.current.logoutAndRedirect();
+      const { result } = renderHook(() => useAuthContext(), {
+        wrapper,
       });
+
+      result.current.logoutAndRedirect();
+      expect(result.current.isAuthenticated).toBe(false);
       await waitFor(() => {
-        expect(hook.current.isAuthenticated).toBe(false);
+        expect(history.location.pathname).toBe("/login");
       });
-      expect(history.location.pathname).toBe("/login");
     });
 
     it("should redirect to the Evergreen /login page otherwise", async () => {
       mockEnv("NODE_ENV", "production");
       const mockFetchPromise = jest.fn().mockResolvedValue({});
       jest.spyOn(global, "fetch").mockImplementation(mockFetchPromise);
 
-      const { Component, hook } = renderComponentWithHook();
-      render(<Component />, { wrapper });
-
-      act(() => {
-        hook.current.logoutAndRedirect();
+      renderHook(() => useAuthContext(), {
+        wrapper,
       });
```

Comment:
I think this test should still be calling `hook.current.logoutAndRedirect`, since the test block is testing this method

Reply:
`logOutAndRedirects` gets called automatically by the `useEffect`. Which is what causes the component to be unmounted on render.

Reply:
The behavior of the component is to automatically redirect you to the login page. `logoutAndRedirect` was never actually causing this component to redirect it was always the redirect from the `useEffect`. If you remove this line from the original test it will still pass.